### PR TITLE
:hammer: Refactor manual_control

### PR DESCRIPTION
**Pull request by MathisFederico**
# Description

A simple refactoring of the manual_control.py file using a new ManualControl object.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [X] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by MathisFederico** [comment]
I'm not sure how to check if I need to modify anything in the documentation

**Comment by rodrigodelazcano** [comment]
@MathisFederico thank you, this looks great. Can you add the `pytest_mock` dependency under [test_requirements.txt](https://github.com/Farama-Foundation/Minigrid/blob/master/test_requirements.txt). 

I think this is the cause of the error in the CI.

**Comment by MathisFederico** [comment]
@rodrigodelazcano I had put it in test_requirements.txt but the setup.py file was badly configured and did not read it !
You might want to switch to [pyproject.toml](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) config instead which is the new standard !

**Comment by rodrigodelazcano** [comment]
@MathisFederico our plan is to transition to pyproject.toml in the future, when we can assure that python developers and users have done a complete transition to it. However, if you wish to make an initial PR for future merging that would be great

File: test_requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
-pytest==7.0.1
\ No newline at end of file
+pytest>=7.0.1
+pytest-mock
```
Comment (rodrigodelazcano) [inline test_requirements.txt:2]: Can you pin the version of pytest-mock to the one you are using?

File: tests/test_envs.py [hunk 2]
```diff
@@ -163,6 +163,13 @@ def test_max_steps_argument(env_spec):
     env.close()
 
 
+@pytest.mark.parametrize("env_id", [spec.id for spec in all_testing_env_specs])
+def test_env_picklable(env_id):
```
Comment (rodrigodelazcano) [inline tests/test_envs.py:167]: Can you revert to your test implementation with pickle dumps and load?

**Comment by rodrigodelazcano** [comment]
Hey @MathisFederico . How are the reviewed changes going? :)

File: test_requirements.txt (hunk 1 above)
Comment (MathisFederico) [reply test_requirements.txt:2]: I used latest whatever it is, I'm always using a fresh virtual environnement

File: tests/test_envs.py (hunk 2 above)
Comment (MathisFederico) [reply tests/test_envs.py:167]: Isn't this just update on not being done yet ? It would be strange that it require an other commit

**Comment by MathisFederico** [comment]
Hey @rodrigodelazcano done !

**Comment by rodrigodelazcano** [comment]
@MathisFederico I think we should remove the extra environment pickable test. Last PR you made already had a great test for this. Can you remove this and also rebase your branch please?

**Review by rodrigodelazcano — APPROVED** [review]

**Pull request merged**